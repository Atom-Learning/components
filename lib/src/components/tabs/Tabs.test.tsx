import { IdProvider } from '@radix-ui/react-id'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Tabs } from '.'

const TabsTest = ({ defaultValue = 'tab1' }) => (
  <IdProvider>
    <Tabs defaultValue={defaultValue}>
      <Tabs.TriggerList>
        <Tabs.Trigger value="tab1">Trigger 1</Tabs.Trigger>
        <Tabs.Trigger value="tab2">Trigger 2</Tabs.Trigger>
      </Tabs.TriggerList>
      <Tabs.Content value="tab1">Important content for tab 1</Tabs.Content>
      <Tabs.Content value="tab2">Important content for tab 2</Tabs.Content>
    </Tabs>
  </IdProvider>
)

const MobileTabsTest = ({ defaultValue = 'tab1' }) => (
  <IdProvider>
    <Tabs defaultValue={defaultValue}>
      <Tabs.TriggerList enableTabScrolling>
        <Tabs.Trigger value="tab1">
          Trigger 1 which is going to be long
        </Tabs.Trigger>
        <Tabs.Trigger value="tab2">
          Trigger 2 which is going to be even longer
        </Tabs.Trigger>
      </Tabs.TriggerList>
      <Tabs.Content value="tab1">Important content for tab 1</Tabs.Content>
      <Tabs.Content value="tab2">Important content for tab 2</Tabs.Content>
    </Tabs>
  </IdProvider>
)

describe('Tabs component', () => {
  it('renders', async () => {
    const { container } = render(<TabsTest />)

    expect(container).toMatchSnapshot()
  })

  it('renders mobile view', async () => {
    const { container } = render(<MobileTabsTest />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<TabsTest />)

    expect(await axe(container)).toHaveNoViolations()
  })

  it('it makes the correct tab content visible, based on the `defaultValue` prop', async () => {
    render(<TabsTest defaultValue="tab2" />)
    const tab1content = await screen.queryByText('Important content for tab 1')
    const tab2content = await screen.queryByText('Important content for tab 2')

    expect(tab1content).not.toBeInTheDocument()
    expect(tab2content).toBeVisible()
  })

  it('allows the user to switch tabs', async () => {
    render(<TabsTest />)

    let tab1content = await screen.queryByText('Important content for tab 1')
    let tab2content = await screen.queryByText('Important content for tab 2')

    expect(tab1content).toBeVisible()
    expect(tab2content).not.toBeInTheDocument()

    await userEvent.click(screen.getByText('Trigger 2'))

    tab1content = await screen.queryByText('Important content for tab 1')
    tab2content = await screen.findByText('Important content for tab 2')

    expect(tab2content).toBeVisible()
    expect(tab1content).not.toBeInTheDocument()
  })

  it("doesn't allow clicking disabled tabs", async () => {
    render(
      <IdProvider>
        <Tabs defaultValue="tab1">
          <Tabs.TriggerList>
            <Tabs.Trigger value="tab1">Trigger 1</Tabs.Trigger>
            <Tabs.Trigger disabled value="tab2">
              Trigger 2
            </Tabs.Trigger>
          </Tabs.TriggerList>
          <Tabs.Content value="tab1">Important content for tab 1</Tabs.Content>
          <Tabs.Content value="tab2">Important content for tab 2</Tabs.Content>
        </Tabs>
      </IdProvider>
    )

    const tab1content = await screen.queryByText('Important content for tab 1')
    const tab2content = await screen.queryByText('Important content for tab 2')

    expect(tab1content).toBeVisible()

    await userEvent.click(screen.queryByText('Trigger 2'))

    expect(tab1content).toBeVisible()
    expect(tab2content).not.toBeInTheDocument()
  })
})
