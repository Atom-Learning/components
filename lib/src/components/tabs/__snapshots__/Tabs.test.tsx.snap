// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders 1`] = `
@media  {
  .c-fIXJvv {
    color: var(--colors-tonal600);
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-fIXJvv > .c-fIXJvv:before,
  .c-fIXJvv > .c-fIXJvv:after {
    display: none;
  }

  .c-bWpZXV {
    background: none;
    border: none;
    cursor: pointer;
    flex-shrink: 0;
    padding: var(--space-4);
    -webkit-user-select: none;
    user-select: none;
    border-bottom: 2px solid transparent;
  }

  .c-bWpZXV[data-state="active"] {
    color: var(--colors-interactive1);
    font-weight: 600;
    letter-spacing: -0.005em;
    border-color: currentColor;
  }

  .c-bWpZXV[data-disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-bWpZXV:not([data-disabled]):hover,
  .c-bWpZXV:not([data-disabled]):focus-visible {
    color: var(--colors-interactive2);
    background: var(--colors-accentA1);
  }

  .c-bWpZXV:not([data-disabled]):active {
    color: var(--colors-interactive3);
  }

  .c-bWpZXV:not([data-disabled]):focus-visible {
    outline: none;
    position: relative;
    z-index: 1;
    box-shadow: white 0px 0px 0px 2px, var(--colors-primary) 0px 0px 0px 4px;
  }

  .c-lmpLrZ {
    flex-shrink: 0;
    display: flex;
    border-bottom: 1px solid var(--colors-base3);
    width: 100%;
  }

  .c-cWMaKf {
    flex-grow: 1;
    font-family: var(--fonts-body);
  }
}

@media  {
  .c-fIXJvv-bndJoy-size-sm {
    font-size: var(--fontSizes-sm);
    line-height: 1.53;
  }

  .c-fIXJvv-bndJoy-size-sm::before {
    content: '';
    margin-bottom: -0.4056em;
    display: table;
  }

  .c-fIXJvv-bndJoy-size-sm::after {
    content: '';
    margin-top: -0.4056em;
    display: table;
  }
}

@media  {
  :root,
  .t-deVgmU {
    --colors-background: white;
    --colors-foreground: #191919;
    --colors-foreground6plus: white;
    --colors-blue1: #F3F8FF;
    --colors-blue2: #E5F1FF;
    --colors-blue3: #D6EAFF;
    --colors-blue4: #C2DEFF;
    --colors-blue5: #B3D4FA;
    --colors-blue6: #057AFF;
    --colors-blue7: #0F67F5;
    --colors-blue8: #194DCC;
    --colors-blue9: #102DA3;
    --colors-blue10: #15235E;
    --colors-blueA1: rgba(15,115,255,0.05);
    --colors-blueA2: rgba(0,115,255,0.1);
    --colors-blueA3: rgba(0,115,255,0.15);
    --colors-blueA4: rgba(0,90,255,0.2);
    --colors-blueA5: rgba(0,96,236,0.27);
    --colors-blueA6: rgba(0,0,255,0.35);
    --colors-blueA7: rgba(0,0,235,0.5);
    --colors-blueA8: rgba(0,1,182,0.7);
    --colors-blueA9: rgba(0,0,140,0.8);
    --colors-blueA10: rgba(9,23,86,0.95);
    --colors-slate1: #F6F6F6;
    --colors-slate2: #EBEBEB;
    --colors-slate3: #E0E0E0;
    --colors-slate4: #CECECE;
    --colors-slate5: #C2C2C2;
    --colors-slate6: #757575;
    --colors-slate7: #545454;
    --colors-slate8: #333333;
    --colors-slate9: #1F1F1F;
    --colors-slate10: #1C1C1C;
    --colors-slateA1: rgba(75,75,75,0.05);
    --colors-slateA2: rgba(55,55,55,0.1);
    --colors-slateA3: rgba(48,48,48,0.15);
    --colors-slateA4: rgba(10,10,10,0.2);
    --colors-slateA5: rgba(29,29,29,0.27);
    --colors-slateA6: rgba(0,0,0,0.35);
    --colors-slateA7: rgba(0,0,0,0.5);
    --colors-slateA8: rgba(0,0,0,0.7);
    --colors-slateA9: rgba(0,0,0,0.8);
    --colors-slateA10: rgba(16,16,16,0.95);
  }

  .interactive-hiContrast1 {
    --colors-interactiveForeground: var(--colors-foreground6plus);
    --colors-interactive1: var(--colors-accent7);
    --colors-interactive2: var(--colors-accent8);
    --colors-interactive3: var(--colors-accent9);
  }

  .accent-blue {
    --colors-accent1: var(--colors-blue1);
    --colors-accentA1: var(--colors-blueA1);
    --colors-accent2: var(--colors-blue2);
    --colors-accentA2: var(--colors-blueA2);
    --colors-accent3: var(--colors-blue3);
    --colors-accentA3: var(--colors-blueA3);
    --colors-accent4: var(--colors-blue4);
    --colors-accentA4: var(--colors-blueA4);
    --colors-accent5: var(--colors-blue5);
    --colors-accentA5: var(--colors-blueA5);
    --colors-accent6: var(--colors-blue6);
    --colors-accentA6: var(--colors-blueA6);
    --colors-accent7: var(--colors-blue7);
    --colors-accentA7: var(--colors-blueA7);
    --colors-accent8: var(--colors-blue8);
    --colors-accentA8: var(--colors-blueA8);
    --colors-accent9: var(--colors-blue9);
    --colors-accentA9: var(--colors-blueA9);
    --colors-accent10: var(--colors-blue10);
    --colors-accentA10: var(--colors-blueA10);
  }
}

<div>
  <div
    class="PJLV"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="c-lmpLrZ PJLV interactive-hiContrast1 accent-blue"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <span
        aria-controls="radix-3-content-tab1"
        aria-selected="true"
        class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-bWpZXV"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="active"
        id="radix-3-trigger-tab1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Trigger 1
      </span>
      <span
        aria-controls="radix-3-content-tab2"
        aria-selected="false"
        class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-bWpZXV"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-3-trigger-tab2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Trigger 2
      </span>
      <span
        aria-controls="radix-3-content-tab3"
        aria-selected="false"
        class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-bWpZXV"
        data-disabled=""
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        disabled=""
        id="radix-3-trigger-tab3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Trigger 3
      </span>
    </div>
    <div
      aria-labelledby="radix-3-trigger-tab1"
      class="c-cWMaKf"
      data-orientation="horizontal"
      data-state="active"
      id="radix-3-content-tab1"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      Important content for tab 1
    </div>
    <div
      aria-labelledby="radix-3-trigger-tab2"
      class="c-cWMaKf"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-3-content-tab2"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-3-trigger-tab3"
      class="c-cWMaKf"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-3-content-tab3"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</div>
`;
